src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2263,6 +2297,72 @@ var projectUniqueNamesForBuildIntToUpdate
+            INuGetProjectContext nuGetProjectContext, 
+            CancellationToken token)
+        {
+            try
+            {
+                nuGetProjectContext.Log(
+                        ProjectManagement.MessageLevel.Info,
+                        Strings.InstallingNetstandard20CompatibilityPackage);
+
+                var resolutionContext = new ResolutionContext(DependencyBehavior.Lowest,
+                includePrelease: false,
+                includeUnlisted: false,
+                versionConstraints: VersionConstraints.None);
+
+                var primarySources = SourceRepositoryProvider.GetRepositories()
+                    .Where(e => e.PackageSource.IsEnabled);
+
+                var log = new LoggerAdapter(nuGetProjectContext);
+
+                // First check for latest stable version of the package.
+                var resolvedPackage = await GetLatestVersionAsync("NETStandard.Library.NETFramework",
+                    nuGetProject,
+                    resolutionContext,
+                    primarySources,
+                    log,
+                    token);
+
+                if (resolvedPackage == null || resolvedPackage.LatestVersion == null)
+                {
+                    // If no stable version of the package could be found, then look for latest pre-release.
+                    resolutionContext = new ResolutionContext(DependencyBehavior.Lowest,
+                        includePrelease: true,
+                        includeUnlisted: false,
+                        versionConstraints: VersionConstraints.None);
+
+                    resolvedPackage = await GetLatestVersionAsync("NETStandard.Library.NETFramework",
+                        nuGetProject,
+                        resolutionContext,
+                        primarySources,
+                        log,
+                        token);
+
+                    if (resolvedPackage == null || resolvedPackage.LatestVersion == null)
+                    {
+                        throw new InvalidOperationException(string.Format(Strings.NoLatestVersionFound, "NETStandard.Library.NETFramework"));
+                    }
+
+                }
+
+                await InstallPackageAsync(nuGetProject,
+                    new PackageIdentity("NETStandard.Library.NETFramework", resolvedPackage.LatestVersion),
+                    resolutionContext,
+                    nuGetProjectContext,
+                    primarySources,
+                    Enumerable.Empty<SourceRepository>(),
+                    token);
+            }
+            catch (InvalidOperationException e) when (e.InnerException.GetType() == typeof(PackageAlreadyInstalledException))
+            {
+                nuGetProjectContext.Log(
+                        ProjectManagement.MessageLevel.Info,
+                        Strings.Netstandard20CompatibilityPackageAlreadyInstalled);
```

Comment:
Why display this? And why not check first if the package is installed? It doesn't seem like this should happen.